cmd/lifecycle/rebaser.go

```diff
@@ -101,7 +102,7 @@ func (r *rebaseCmd) Exec() error {
 		Logger:      cmd.DefaultLogger,
 		PlatformAPI: r.PlatformAPI,
 	}
-	report, err := rebaser.Rebase(r.appImage, newBaseImage, r.AdditionalTags)
+	report, err := rebaser.Rebase(r.appImage, newBaseImage, r.AdditionalTags, r.OutputImageRef)
```

Comment:
What's the thought behind not executing `r.appImage.Rename(outputImageRef)` at this level?

I worry a bit about the `r.appImage` getting `Save`'d before a rename. It is supposed to be untouched if given as the `-previous-image`.

Reply:
It's not here because some of the work inside rebaser.Rebase gets meta data from the `appImage`. 

Specifically, when we call [image.DecodeLabel](https://github.com/buildpacks/lifecycle/blob/50646839c80cc899193a8ed587ad59a20ec3f15a/rebaser.go#L30) inside that method, that in turn calls [image.Found](https://github.com/buildpacks/imgutil/blob/main/image.go#L64).

Here's the description for image.Found:

> Found tells whether the image exists in the repository by `Name()`.

So if we're solving this by using rename, we cannot rename until after [image.DecodeLabel](https://github.com/buildpacks/lifecycle/blob/50646839c80cc899193a8ed587ad59a20ec3f15a/rebaser.go#L30).

Reply:
Maybe we create a new method that is something like `image.DecodeLabelNoValidation`?

Reply:
That makes sense.

It feels odd that we have to pass in `AdditionalTags` and `OutputImageRef` and that we call `Save` on `appImage`, which today has `OutputImageRef` as the name already.

I wonder if we could make the signature more like 

```golang
func (r *Rebaser) Rebase(previousImage imgutil.Image, newBaseImage imgutil.Image, exportTargets []string) (RebaseReport, error) { {}
```

We could `Rename(exportTargets[0])` and then `Save(exportTargets[1:]...)`.

I'm still not in love the `Rename` flow - I would like to investigate your first approach before fully committing to this. In theory using the `remote.FromBaseImage(previousImage)` with an `outputImageRef` seems possible.